numpy/core/src/umath/ufunc_object.c

```diff
@@ -2089,6 +2107,124 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         goto fail;
     }
 
+    if (axis) {
+        /*
+         * possibly remap axes.
+         */
+        int axis_rqst;
```

Comment:
Ugh, you're right. I did that because that's what we used elsewhere in the public API, not because it was the right decision.